source/src/protocols/motifs/LigandDiscoverySearch.cc

```diff
@@ -292,694 +366,1790 @@ utility::vector1<core::conformation::ResidueOP> LigandDiscoverySearch::get_all_r
+							continue;
+						}
+					}
+
+					//append ligand to minipose for early scoring
+					minipose->append_residue_by_jump(*ligresOP, 1);
+
+					//set up values to be passed into score_minipose and used downstream
+					core::Real fa_rep;
+					core::Real fa_atr;
+					core::Real fa_atr_rep_score_before;
+
+					//score the minipose and set the value to a bool
+					bool minipose_scoring = score_minipose(minipose,fa_rep,fa_atr,fa_atr_rep_score_before);
+
+					//delete the last residue off minipose (the ligand), since we no longer need it and can recycle the minipose for further iterations
+					minipose->delete_residue_slow(minipose->size());
+
+					//if the value is true, the minipose passes initial scoring
+					//if false, the placement can be killed
+					if ( !minipose_scoring ) {
+						++clashing_counter;
+						continue;
+					}
+
+					++passed_placement_counter;
+
+					//append ligand to working pose
+					//using version of append to put it in a new chain
+					working_pose_->append_residue_by_jump(*ligresOP, working_pose_->size(), "", "", true);
+
+					//apply constraint set to working_pose_
+					add_constraints_to_working_pose(trip_atom_1, trip_atom_2, trip_atom_3, working_position, ligresOP);
+
+					//get free energy of pose with placed ligand before highresdock
+					//declaration of variable
+					core::Real delta_score = 10000;
+
+					//use fa atr/rep or whole function based on highresdock_with_whole_score_fxn flag to get ddg before highresdock
+					if ( option[ OptionKeys::motifs::highresdock_with_whole_score_fxn ] ) {
+						//whole
+						delta_score = get_pose_ddg(whole_score_fxn_, working_pose_);
+					} else {
+						//atrrep
+						delta_score = get_pose_ddg(fa_atr_rep_fxn_, working_pose_);
+					}
+
+					ms_tr.Debug << "Pre-move delta score = " << delta_score << ", fa_atr = " << fa_atr << ", fa_rep = " << fa_rep << ", fa_atr_rep before = " << fa_atr_rep_score_before << std::endl;
+
+					//attempt to use movers to optimize placement a little more
+					//begin setting up objects to make HighResDocker mover work
+
+					//add ligand to pose residue type set if it is not already in the set
+					//program doesn't work if the ligand isn't added to the pose residue type sets
+					//also add it to the original pdb
+					if ( ligand_added == false ) {
+						ligand_added = true;
+
+						//add the ligand mrt to the working and original poses
+						add_ligand_to_pose_residuetypeset(lig_mrt);
+					}
+
+					//apply the highresdocker to working_pose
+					run_HighResDock_on_working_pose(my_HighResDocker);
+
+					//use fa atr/rep or whole function based on highresdock_with_whole_score_fxn flag to get ddg before highresdock
+					if ( option[ OptionKeys::motifs::highresdock_with_whole_score_fxn ] ) {
+						//whole
+						delta_score = get_pose_ddg(whole_score_fxn_, working_pose_);
+					} else {
+						//atrrep
+						delta_score = get_pose_ddg(fa_atr_rep_fxn_, working_pose_);
+					}
```

Comment:
Might simplify things if you have something like a `working_score_fxn_` ScoreFunctionOP member variable, and set that at initialization time to either `whole_score_fxn_` or `fa_atr_rep_fxn_` based on the option. Then you can just use `working_score_fxn_` directly, without having to re-consult the option in multiple locations.